### ISSUE TITLE
500 error page is blank

### ISSUE THREAD

**Issue by muffinresearch** [mozilla/addons-frontend#1880]
### Describe the problem and steps to reproduce it:

On addons-dev go to: /en-GB/firefox/users/edit/
Note the blank grey page.

### What happened?

A blank 500 error page. The traceback on the server is:

```
[1] [2017-02-23T11:24:01.392Z] ERROR: server/71328 on hanuman.local: "users" not found in API_ADDON_TYPES_MAPPING (app=amo)
[1]     AddonTypeNotFound: "users" not found in API_ADDON_TYPES_MAPPING
[1]         at apiAddonType (/Users/Whatevs/code/addons-frontend/src/core/utils.js:176:11)
[1]         at Object.loadByCategoryIfNeeded [as promise] (/Users/Whatevs/code/addons-frontend/src/core/searchUtils.js:112:16)
[1]         at /Users/Whatevs/code/addons-frontend/node_modules/redux-connect/lib/helpers/utils.js:184:38
[1]         at Array.reduce (native)
[1]         at _loop3 (/Users/Whatevs/code/addons-frontend/node_modules/redux-connect/lib/helpers/utils.js:182:46)
[1]         at /Users/Whatevs/code/addons-frontend/node_modules/redux-connect/lib/helpers/utils.js:200:7
[1]         at iterateOverResults (/Users/Whatevs/code/addons-frontend/node_modules/redux-connect/lib/helpers/utils.js:53:12)
[1]         at process._tickCallback (internal/process/next_tick.js:103:7)
```

There's also `Unhandled promise rejection` errors in the console.

### What did you expect to happen?

A missing addon type should probably be a 404. But also the 500 error page should be being displayed correctly.

Looks like this:

<img width="350" alt="screen shot 2017-02-22 at 10 10 50 am" src="https://cloud.githubusercontent.com/assets/211578/23220477/c51ce9c0-f8e7-11e6-8689-d1f636249fb0.png">

**Comment by kumar303** [comment]
redux-connect strikes again. Whenever one of our callbacks accidentally throws an error instead of returning a promise, that error [is not caught by redux-connect](https://github.com/makeomatic/redux-connect/blob/master/modules/helpers/utils.js#L133) so it disappears on the event loop.

I have an idea for a fix that makes sure our callbacks never do that. 

However, it would be great if we could handle random errors on the event loop somehow. I guess that means hooking into [process.on('uncaughtException')](https://nodejs.org/api/process.html#process_event_uncaughtexception) and then somehow figuring out how to turn that into a 500 page. Maybe [connect-domain](https://github.com/baryshev/connect-domain) would help? I'm going to fix the easy part first.

**Comment by kumar303** [comment]
Filed upstream but we can work around it: https://github.com/makeomatic/redux-connect/issues/108

**Comment by ValentinaPC** [comment]
Verified as fixed on AMO-dev FF51(Android 6.0.1)
 - While loading this: https://addons-dev.allizom.org/en-US/firefox/users/edit/  on device, a 500 error page is displayed.  
Postfix screenshot:
![an](https://cloud.githubusercontent.com/assets/15685960/23354387/a71b15fa-fcd9-11e6-8745-32d794cc71d3.png)

Also, tried other pages that don't have correspondent in mobile site and obtained **different** results:
- https://addons-dev.allizom.org/en-US/firefox/seamonkey/ - 404 error page
- https://addons-dev.allizom.org/en-US/firefox/language-tools/ - correspondent desktop page
- https://addons-dev.allizom.org/en-US/firefox/complete-themes/ - strange page
![ct](https://cloud.githubusercontent.com/assets/15685960/23354611/beba2e16-fcda-11e6-8e7d-5c29faa04d38.png)

@muffinresearch : please let me know if anything else should be tested here?
I'll log a separate issue for the last case. Thanks!
